pandas/tests/io/json/test_pandas.py

```diff
@@ -1597,3 +1597,10 @@ def test_json_indent_all_orients(self, orient, expected):
     def test_json_negative_indent_raises(self):
         with pytest.raises(ValueError, match="must be a nonnegative integer"):
             pd.DataFrame().to_json(indent=-1)
+
+    def test_emca_262_nan_inf_support(self):
```

Comment:
can you add the strings, 'NaN', '-Infinity', and 'Infinity' to be sure we are not over parsing. also do we parse more common 'Inf' and '-Inf' ?

Reply:
Good idea. So note that I also had to add an "a" to this as the parsing worked correctly, but a subsequent `astype("float64")` that the JSON code does would turn the strings into their float representation.

W.r.t "Inf" and "-Inf" the upside to not doing them is that they aren't part of [EMCA 262](https://www.ecma-international.org/ecma-262/10.0/index.html#Title) so we could point to that document if we don't include. The one gap from EMCA 262 that we don't have is "+Infinity" which I am happy to add here